setup.py

```diff
@@ -15,7 +15,7 @@
     maintainer_email="aesara-devs@gmail.com",
     packages=["aemcmc"],
     install_requires=[
-        "numpy>=1.18.1",
+        "numpy>=1.21.0",
```

Comment:
Wont this cause problems when making a release since we support python 3.7? As far as I'm aware python 3.7 supports numpy 1.17+.

We might want to decide if we want to follow the standard described in [NEP29](https://numpy.org/neps/nep-0029-deprecation_policy.html)

Reply:
`aeppl` does not have this problem, even though [the version that is specified[(https://github.com/aesara-devs/aeppl/blob/main/setup.py) in `setup.py` is `1.18.1`. Nothing else is specified in the tests.

Reply:
The error happens on [this line](https://github.com/aesara-devs/aemcmc/blob/165056aa5f9e4f8d005bcef668061f08c543bae9/.github/workflows/test.yml#L101) and I would like why it fails in `aemcmc`'s CI while it does not in `aeppl`'s.

Reply:
I think I found the culprit. The error actually happens in this [line](https://github.com/aesara-devs/aemcmc/blob/165056aa5f9e4f8d005bcef668061f08c543bae9/.github/workflows/test.yml#L102), not the one you linked to. If you look at the top of the error message it mentions a pip subprocess that failed to install build dependencies for one of the packages in the `requirements.txt` file. This package so happens to be `polyagamma`. It attempts to build `numpy 1.19` from source since its not supported by python 3.10 and then fails.

Reply:
I've been working on a PR to put out 3.10 wheels but I keep getting weird CI failures that weren't there before.

Reply:
Finally got it to work....i think

Reply:
Having trouble pushing out a release. It seems now all of a sudden building wheels on Windows with python 3.8 causes the import of the package to fail. See: https://github.com/zoj613/polyagamma/runs/5841514828?check_suite_focus=true#step%3A7%3A318=

I opened an issue at https://github.com/pypa/cibuildwheel/issues/1076

Reply:
Okay, i've managed to publish v1.3.3 for `polyagamma` after a long debugging process. See this [comment](https://github.com/zoj613/polyagamma/pull/110#issue-1194466794) for a summary. @rlouf could you try out the CI pipeline again, ensuring that the latest version is installed so it works for 3.10?